crates/sui-storage/src/event_store/mod.rs

```diff
@@ -126,13 +129,16 @@ trait EventStore {
         end_time: u64,
         module: ModuleId,
         limit: usize,
-    ) -> Result<Self::EventIt, EventStoreError>;
+    ) -> Result<Self::EventIt, SuiError>;
 }
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum EventStoreError {
+    #[error("Generic EventStoreError: {0:?}")]
     GenericError(Box<dyn std::error::Error>),
+    #[error("sqlx error: {0:?}")]
     SqlError(sqlx::Error),
+    #[error("Specified limit is too high: {0}")]
```

Comment:
show the allowed max?

Reply:
It turns out there is another error that I can use that already has that.